docs/admin-guide/admin-configuration.md

```diff
@@ -413,6 +414,54 @@ The following table lists the configurable attributes for enhanced search.
 | `cve`            | Extends enhanced search to allow searching of Common Vulnerabilities and Exposures (CVE).                                                  |
 | `updateInterval` | Sets the interval at which the searchable database of CVE items is refreshed.                                                              |
 
+<a name="userprefs_config"></a>
+
+## Setting user preferences
+
+The user preferences extension provides an API endpoint for adding configurable user preferences for a repository. This custom extension, not a part of the OCI distribution, is accessible only by authenticated users of the registry. Unauthenticated users are denied access.
+
+To allow users to set preferences, add the `userprefs` attribute under `extensions` in the configuration file.
+
+The `userprefs` endpoint accepts as a query parameter an `action` to perform along with any other required parameters for the specified action. The actions currently implemented do not require an HTTP payload, nor do they return any related data other than an HTTP response code.
+
+### Current functionality
+
+The current functions implemented by this extension include:
+
+- Toggling the star (favorites) icon for a repository.
+- Toggling the bookmark icon for a repository.
+
+#### Toggle repository star
+
+This action sets the repository star property to `true` if it is `false`, and to `false` if it is `true`.
+
+| Action | Parameter | Parameter Description |
+| --- | --- | --- |
+| toggleStar | repository | The name of the repository whose star is to be changed |
+
+This example toggles a star on a repository named repoName:
+
+```
+PUT
+http://localhost:5000/v2/_zot/ext/userprefs?
+action=toggleStar&repo=repoName
+```
+
+#### Toggle repository bookmark
+
+This action sets the repository bookmark property to `true` if it is `false`, and to `false` if it is `true`.
+
+| Action | Parameter | Parameter Description |
+| --- | --- | --- |
+| toggleBookmark | repository | The name of the repository whose bookmark is to be changed |
```

Comment:
Should we keep under the `Parameter` column the actual name of the parameter used in the query params and explain it's meaning under `Parameter Description`?
```suggestion
| toggleBookmark | repo| The name of the repository whose bookmark is to be changed |
```

This would apply to the other example regarding `toggleStars` as well

Reply:
Corrections made.